icechunk-python/python/icechunk/_icechunk_python.pyi

```diff
@@ -253,15 +229,47 @@ class StaticCredentials:
         session_token: str | None = None,
     ): ...
 
-class Credentials:
+class S3Credentials:
     class FromEnv:
         def __init__(self) -> None: ...
 
     class DontSign:
         def __init__(self) -> None: ...
 
     class Static:
-        def __init__(self, _0: StaticCredentials) -> None: ...
+        def __init__(self, _0: S3StaticCredentials) -> None: ...
+
+class Credentials:
+    class S3:
+        def __init__(self, credentials: S3Credentials) -> None: ...
+
+class Storage:
+    """Storage configuration for an IcechunkStore
+
+    Currently supports memory, filesystem, and S3 storage backends.
+    Use the class methods to create a StorageConfig object with the desired backend.
+
+    Ex:
+    ```
+    storage_config = StorageConfig.memory("prefix")
+    storage_config = StorageConfig.filesystem("/path/to/root")
+    storage_config = StorageConfig.object_store("s3://bucket/prefix", vec!["my", "options"])
+    storage_config = StorageConfig.s3_from_env("bucket", "prefix")
+    storage_config = StorageConfig.s3_from_config("bucket", "prefix", ...)
+    ```
+    """
+
+    @staticmethod
+    def make_s3(
```

Comment:
I personally don't love the naming of `make_` and would prefer just simply `s3()`, `in_memory`, but I am open to alternatives if others like verbs

Reply:
done